shapely/_geometry_helpers.pyx

```diff
@@ -214,7 +214,7 @@ cdef const GEOSGeometry* GetRingN(GEOSContextHandle_t handle, GEOSGeometry* poly
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def get_parts(object[:] array, bint extract_rings=0):
+def get_parts(np.ndarray[object] array, bint extract_rings=0):
```

Comment:
typically when working with memoryviews, we would make this `const object[:]`, like was done in https://github.com/shapely/shapely/pull/1744, but that doesn't work for object dtype. Using the numpy ndarray syntax allows read-only by default